service/worker/archiver/activities.go

```diff
@@ -110,9 +112,12 @@ func deleteHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 			err = temporal.NewNonRetryableApplicationError(err.Error(), "", nil)
 		}
 	}()
-	err = container.HistoryV2Manager.DeleteHistoryBranch(ctx, &persistence.DeleteHistoryBranchRequest{
-		BranchToken: request.BranchToken,
-		ShardID:     request.ShardID,
+	_, err = container.OperatorClient.DeleteWorkflowExecution(ctx, &operatorservice.DeleteWorkflowExecutionRequest{
```

Comment:
When this is called, before this PR, the mutable state is already gone.
I think operator DeleteWorkflow depends on mutable state. If mutable state is gone, the history won't be deleted.

Reply:
This is where we need to sync offline. In the delete manager, it will return if the workflow should be archived. So the mutable state will not be deleted before the archival.

Reply:
Changed to use history client